From b1a0e5d9cd56732caea077e66d5b019129a54394 Mon Sep 17 00:00:00 2001
From: Eric Miao <eric.y.miao@gmail.com>
Date: Wed, 24 Oct 2012 17:25:29 +0800
Subject: [PATCH 3/3] trace-cmd: Add support for python GUI plugins

Signed-off-by: Eric Miao <eric.y.miao@gmail.com>
---
 Makefile |   13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 85fdff9..6adfe70 100644
--- a/Makefile
+++ b/Makefile
@@ -83,8 +83,8 @@ PYTHON_GUI	:= ctracecmd.so ctracecmdgui.so
 # Can build python?
 ifeq ($(shell sh -c "python-config --includes > /dev/null 2>&1 && echo y"), y)
 	PYTHON_PLUGINS := plugin_python.so
-	BUILD_PYTHON := $(PYTHON) $(PYTHON_PLUGINS)
-	PYTHON_SO_INSTALL := ctracecmd.install
+	BUILD_PYTHON := $(PYTHON_GUI) $(PYTHON_PLUGINS)
+	PYTHON_SO_INSTALL := ctracecmd.install ctracecmdgui.install
 	PYTHON_PY_INSTALL := event-viewer.install tracecmd.install tracecmdgui.install
 endif
 endif # NO_PYTHON
@@ -376,6 +376,11 @@ libparsevent.a: $(PEVENT_LIB_OBJS)
 $(TCMD_LIB_OBJS): %.o: $(src)/%.c
 	$(Q)$(do_fpic_compile)
 
+ifeq ($(BUILDGUI),1)
+$(TRACE_VIEW_OBJS): %.o: $(src)/%.c
+	$(Q)$(do_fpic_compile)
+endif
+
 libtracecmd.so: $(TCMD_LIB_OBJS)
 	$(Q)$(do_compile_shared_library)
 
@@ -549,10 +554,10 @@ ctracecmd.so: $(TCMD_LIB_OBJS) ctracecmd.i
 	gcc -fpic -c $(PYTHON_INCLUDES)  ctracecmd_wrap.c
 	$(CC) --shared $(TCMD_LIB_OBJS) ctracecmd_wrap.o -o ctracecmd.so
 
-ctracecmdgui.so: $(TRACE_VIEW_OBJS) $(LIB_FILE)
+ctracecmdgui.so: $(TRACE_VIEW_OBJS) $(LIB_FILE) ctracecmdgui.i
 	swig -Wall -python -noproxy ctracecmdgui.i
 	gcc -fpic -c  $(CFLAGS) $(INCLUDES) $(PYTHON_INCLUDES) $(PYGTK_CFLAGS) ctracecmdgui_wrap.c
-	$(CC) --shared $^ $(LIBS) $(CONFIG_LIBS) ctracecmdgui_wrap.o -o ctracecmdgui.so
+	$(CC) --shared $(TRACE_VIEW_OBJS) $(LIBS) $(CONFIG_LIBS) ctracecmdgui_wrap.o -o ctracecmdgui.so
 
 PHONY += python
 python: $(PYTHON)
-- 
1.7.9.5

